code/game/machinery/bots/medbot.dm

```diff
@@ -472,47 +472,54 @@
 			spawn(INJECTION_TIME)
 				inject_patient()
 
-//	src.speak(reagent_id)
-	reagent_id = null
-	return
-
 /obj/machinery/bot/medbot/proc/inject_patient()
-	if ((get_dist(src, src.patient) <= 1) && (src.on))
-		if((reagent_id == "internal_beaker") && (src.reagent_glass) && (src.reagent_glass.reagents.total_volume))
-			src.reagent_glass.reagents.trans_to(src.patient,src.injection_amount) //Inject from beaker instead.
-			src.reagent_glass.reagents.reaction(src.patient, 2)
+	if ((get_dist(src, patient) <= 1) && (on))
+		if((reagent_id == "internal_beaker") && (reagent_glass) && (reagent_glass.reagents.total_volume))
+			reagent_glass.reagents.trans_to(patient,injection_amount) //Inject from beaker instead.
+			reagent_glass.reagents.reaction(patient, 2)
 		else
-			src.patient.reagents.add_reagent(reagent_id,src.injection_amount)
-		visible_message("<span class='danger'>[src] injects [src.patient] with the syringe!</span>")
+			patient.reagents.add_reagent(reagent_id,injection_amount)
+
+		visible_message("<span class='danger'>[src] injects [patient] with the syringe!</span>")
 
-	src.icon_state = "[src.icon_initial][src.on]"
-	src.currently_healing = 0
+	icon_state = "[icon_initial][on]"
+	currently_healing = 0
+	reagent_id = null
```

Comment:
Primary change one - Upon injection, the reagent_id is set to null, which forces the medbot to re-evaluate the patient, which in turn leads to them not constantly injecting nothing